cpp/src/arrow/python/builtin_convert.cc

```diff
@@ -380,44 +385,55 @@ class TypedConverter : public SeqConverter {
 template <typename BuilderType, class Derived>
 class TypedConverterVisitor : public TypedConverter<BuilderType> {
  public:
-  Status AppendData(PyObject* obj, int64_t size) override {
+  Status AppendSingle(PyObject* obj) override {
+    // static_cast<> below devirtualizes the method calls (for performance?)
```

Comment:
This is the CRTP; if you call the method from the derived class I don't know whether the call is still virtual or not. I doubt the performance difference is enough to really matter.